x-pack/platform/packages/shared/kbn-slo-schema/src/schema/slo.ts

```diff
@@ -77,29 +77,40 @@ function isValidId(id: string): boolean {
   return validLength && /^[a-z0-9-_]+$/.test(id);
 }
 
-const sloDefinitionSchema = t.intersection([
-  t.type({
-    id: sloIdSchema,
-    name: t.string,
-    description: t.string,
-    indicator: indicatorSchema,
-    timeWindow: timeWindowSchema,
-    budgetingMethod: budgetingMethodSchema,
-    objective: objectiveSchema,
-    settings: settingsSchema,
-    revision: t.number,
-    enabled: t.boolean,
-    tags: tagsSchema,
-    createdAt: dateType,
-    updatedAt: dateType,
-    groupBy: groupBySchema,
-    version: t.number,
-  }),
-  t.partial({
-    createdBy: t.string,
-    updatedBy: t.string,
-  }),
-]);
+const requiredSloFields = t.type({
+  id: sloIdSchema,
+  name: t.string,
+  description: t.string,
+  indicator: indicatorSchema,
+  timeWindow: timeWindowSchema,
+  budgetingMethod: budgetingMethodSchema,
+  objective: objectiveSchema,
+  settings: settingsSchema,
+  revision: t.number,
+  enabled: t.boolean,
+  tags: tagsSchema,
+  createdAt: dateType,
+  updatedAt: dateType,
+  groupBy: groupBySchema,
+  version: t.number,
+});
+
+const optionalSloFields = t.partial({
+  createdBy: t.string,
+  updatedBy: t.string,
+});
+
+const baseSloSchema = t.intersection([requiredSloFields, optionalSloFields]);
+
+const dashboardsWithIdSchema = t.partial({ dashboards: t.array(t.type({ id: t.string })) });
+const dashboardsWithRefIdSchema = t.partial({ dashboards: t.array(t.type({ refId: t.string })) });
+
+const artifactsWithIdSchema = t.partial({ artifacts: dashboardsWithIdSchema });
```

Comment:
Eventually artifacts could contain more things other than dashboards, so maybe renaming it to something like `artifactsWithDashboardIdSchema` makes more sense? I wouldn't block the PR for this, if you feel it is a quick change and makes sense, feel free to update.